File: backend/app/schemas/inquiry.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class InquiryCreate(BaseModel):
    name: str
    phone: str
    wechat: Optional[str] = None
    email: Optional[str] = None
    product: Optional[str] = None  # 改为 product 以匹配 Inquiry 模型
    message: str


class InquiryUpdate(BaseModel):
    name: Optional[str] = None
    phone: Optional[str] = None
    wechat: Optional[str] = None
    email: Optional[str] = None
    company: Optional[str] = None
    message: Optional[str] = None
    status: Optional[str] = None
    is_active: Optional[bool] = None


class InquiryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: str
    name: str
    phone: str
    wechat: Optional[str] = None
    email: Optional[str] = None
    company: Optional[str] = None
    message: str
    status: str
    is_active: bool = True
    created_at: datetime
    updated_at: datetime

File: frontend/stores/inquiry.ts
import { defineStore } from 'pinia';
import { useApi } from '~/composables/useApi';

export interface Inquiry {
  id: string;
  name: string;
  company: string | null;
  phone: string | null;
  email: string | null;
  product: string | null;
  quantity: string | null;
  message: string | null;
  status: string;
  is_active: boolean;
  created_at: string;
}

export const useInquiryStore = defineStore('inquiry', {
  state: () => ({
    inquiries: [] as Inquiry[],
    currentInquiry: null as Inquiry | null,
    loading: false,
    error: null as string | null,
  }),
  getters: {
    activeInquiries: (state) => state.inquiries.filter((i) => i.is_active),
    pendingInquiries: (state) => state.inquiries.filter((i) => i.status === 'pending'),
  },
  actions: {
    async fetchInquiries() {
      this.loading = true;
      this.error = null;
      try {
        const { request } = useApi();
        const response = await request<{ items: Inquiry[]; total: number }>('/inquiries');
        this.inquiries = response.items || [];
      } catch (e: any) {
        this.error = e.message;
      } finally {
        this.loading = false;
      }
    },
    async fetchInquiry(id: string) {
      this.loading = true;
      this.error = null;
      try {
        const { request } = useApi();
        this.currentInquiry = await request<Inquiry>(`/inquiries/${id}`);
      } catch (e: any) {
        this.error = e.message;
      } finally {
        this.loading = false;
      }
    },
    async createInquiry(data: Partial<Inquiry>) {
      this.loading = true;
      this.error = null;
      try {
        const { request } = useApi();
        await request('/inquiries', { method: 'POST', body: data, skipAuth: true });
      } catch (e: any) {
        this.error = e.message;
      } finally {
        this.loading = false;
      }
    },
    async updateInquiry(id: string, data: Partial<Inquiry>) {
      this.loading = true;
      this.error = null;
      try {
        const { request } = useApi();
        await request(`/inquiries/${id}`, { method: 'PUT', body: data });
        await this.fetchInquiries();
      } catch (e: any) {
        this.error = e.message;
      } finally {
        this.loading = false;
      }
    },
    async deleteInquiry(id: string) {
      this.loading = true;
      this.error = null;
      try {
        const { request } = useApi();
        await request(`/inquiries/${id}`, { method: 'DELETE' });
        await this.fetchInquiries();
      } catch (e: any) {
        this.error = e.message;
      } finally {
        this.loading = false;
      }
    },
  },
});
